Benchmarks/BenchmarkReporter.swift

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Protocol that defines how are the benchmark results printed to the stdout. 
+protocol BenchmarkReporter {
```

Comment:
How about just "printer" instead of "reporter"?
```suggestion
protocol BenchmarkPrinter {
```

Same for `JSONBenchmarkPrinter` and `PlainTextBenchmarkPrinter` below.

Reply:
Edit: resolving this comment in favor of some bigger code refactoring!

Reply:
The one concern I have with naming it a printer is that it probably won't always print items. In one pathway, we might need to directly call into Python to provide the appropriate benchmark values to an external framework instead of logging them to the console. In that case, it would be reporting to an external API, and in the printing case it could be seen as reporting them to the console.

Reply:
I see. My "printer" comments are obsolete after https://github.com/tensorflow/swift-models/pull/230#pullrequestreview-322493790: removing `BenchmarkReporter` and move printing functionality to `BenchmarkResults.print(using:)`.

Additional functionality (e.g. calling into Python) can be defined as extension methods on `BenchmarkResults` or top-level functions.